source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -822,14 +822,15 @@ FilterConfig::FilterConfig(const envoy::extensions::filters::http::lua::v3::Lua&
 FilterConfigPerRoute::FilterConfigPerRoute(
     const envoy::extensions::filters::http::lua::v3::LuaPerRoute& config,
     Server::Configuration::ServerFactoryContext& context)
-    : disabled_(config.disabled()), name_(config.name()) {
+    : disabled_(config.disabled()), name_(config.name()), filter_context_(config.filter_context()) {
   if (disabled_ || !name_.empty()) {
-    return;
+    return; // Filter is disabled or explicit script name is provided.
+  } else if (config.has_source_code()) {
```

Comment:
Because there's a `return` in the above block
```suggestion
  }
  if (config.has_source_code()) {
```